src/Lykke.Logs/Loggers/LykkeConsole/BufferedConsoleLogMessageWriterDecorator.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using AsyncFriendlyStackTrace;
+using Microsoft.Extensions.Logging.Console.Internal;
+
+namespace Lykke.Logs.Loggers.LykkeConsole
+{
+    internal class BufferedConsoleLogMessageWriterDecorator : IConsoleLogMessageWriter
+    {
+        private const int MaxQueuedMessages = 1024 * 4;
+        
+        private readonly IConsoleLogMessageWriter _inner;
+        private readonly bool _disposeInner;
+
+        private readonly BlockingCollection<LogMessageEntry> _messageQueue;
+        private readonly Thread _outputThread;
+        
+        public BufferedConsoleLogMessageWriterDecorator(IConsoleLogMessageWriter inner, bool disposeInner = true)
+        {
+            _inner = inner ?? throw new ArgumentNullException(nameof(inner));
+            _disposeInner = disposeInner;
+
+            _messageQueue = new BlockingCollection<LogMessageEntry>(MaxQueuedMessages);
+            
+            _outputThread = new Thread(ProcessLogQueue)
+            {
+                IsBackground = true,
+                Name = "Console logger queue processing thread"
+            };
+            _outputThread.Start();
+        }
+
+        public void Write(LogMessageEntry entry)
+        {
+            if (!_messageQueue.IsAddingCompleted)
```

Comment:
Any lock? To prevent parallel writes to _inner.

Reply:
ConsoleLogMessageWriter supports parallel writes themself